starknet-ff/Cargo.toml

```diff
@@ -10,26 +10,29 @@ homepage = "https://starknet.rs/"
 description = """
 Starknet field element type
 """
-keywords = ["ethereum", "starknet", "web3"]
+keywords = ["ethereum", "starknet", "web3", "no_std"]
 
 [package.metadata.docs.rs]
 all-features = true
 
 [dependencies]
-ark-ff = "0.3.0"
-bigdecimal = { version = "0.3.0", optional = true }
-crypto-bigint = "0.4.9"
-hex = "0.4.3"
-num-bigint = { version = "0.4.3", optional = true }
-serde = "1.0.152"
-thiserror = "1.0.30"
+ark-ff = { version = "0.3.0", default-features = false }
+crypto-bigint = { version = "0.4.9", default-features = false }
+hex = { version = "0.4.3", default-features = false }
+serde = { version = "1.0.152", optional = true, default-features = false }
+bigdecimal-no-std = { version = "0.4.0", optional = true, default-features = false }
```

Comment:
As mentioned in the Telegram chat, it looks like `bigdecimal-no-std` isn't _really_ `no_std` compatible as its name suggests - I still have to enable `std` for it to compile for `thumbv6m-none-eabi`.

Maybe we simply revert back to using the original `bigdecimal` crate and make it depend on `std` instead?

Reply:
> Maybe we simply revert back to using the original bigdecimal crate and make it depend on std instead?

Done. Good enough for now.
I will try to create a `no_std` branch for `bigdecimal` later, and maybe merge it in the original repo. But it is not really active...